rpc-server/src/modules/blocks/methods.rs

```diff
@@ -1,51 +1,253 @@
 use crate::config::ServerContext;
 use crate::errors::RPCError;
 use crate::modules::blocks::utils::{
-    fetch_block_from_s3, fetch_chunk_from_s3, fetch_shard_from_s3, is_matching_change,
-    scylla_db_convert_block_hash_to_block_height,
+    fetch_block_from_cache_or_get, fetch_block_from_s3, fetch_chunk_from_s3, fetch_shard_from_s3,
+    is_matching_change, scylla_db_convert_block_hash_to_block_height,
     scylla_db_convert_chunk_hash_to_block_height_and_shard_id,
 };
-use crate::utils::proxy_rpc_call;
 #[cfg(feature = "shadow_data_consistency")]
 use crate::utils::shadow_compare_results;
 use jsonrpc_v2::{Data, Params};
 
 use near_primitives::trie_key::TrieKey;
 use near_primitives::views::StateChangeValueView;
 
+#[allow(unused_mut)]
+#[cfg_attr(feature = "tracing-instrumentation", tracing::instrument(skip(data)))]
+pub async fn block(
+    data: Data<ServerContext>,
+    Params(mut params): Params<near_jsonrpc_primitives::types::blocks::RpcBlockRequest>,
+) -> Result<near_jsonrpc_primitives::types::blocks::RpcBlockResponse, RPCError> {
+    tracing::debug!("`block` called with parameters: {:?}", params);
+
+    // Increase the OPTIMISTIC_REQUESTS_TOTAL metric if the request has optimistic finality.
+    if let near_primitives::types::BlockReference::Finality(finality) = &params.block_reference {
+        // Finality::None stands for optimistic finality.
+        if finality == &near_primitives::types::Finality::None {
+            crate::metrics::OPTIMISTIC_REQUESTS_TOTAL.inc();
+        }
+    }
+    crate::metrics::BLOCK_REQUESTS_TOTAL.inc();
+
+    #[cfg(not(feature = "shadow_data_consistency"))]
```

Comment:
Since we decided that we're going to respond with the response Read RPC has crafted, I'd want to get rid of redundant `#[cfg(not(feature = "shadow_data_consistency"))]` here.

You just don't need to put all the logic of the feature into the `let result = { .. }`, but just do it in the `{ .. }`. Our goal is to print a log with the shadow data check result. We don't have to return any results from there. Please, see the `query` for a reference.

Reply:
And you did it as I am asking in the `chunk` method below.